### ISSUE TITLE
NGINX: unknown hash function used by peer

### ISSUE THREAD

**Issue by SergeyFrolov** [SergeyFrolov/gotapdance#24]
In 5b8d806c893ec880597742e25582f7cde9e1921b (with enabled TLS verification) I get `establishTLStoDecoy(tapdance2.freeaeskey.xyz,192.122.190.105:443) failed with tls: unknown hash function used by peer`. Does ztls support SHA512? Because the only difference between tapdance1(works) and tapdance2(doesn't) I found is SHA512:

**tapdance1.freeaeskey.xyz**:
```
Cipher Suite: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)
Signature Hash Algorithm: 0x0401
    Signature Hash Algorithm Hash: SHA256 (4)
    Signature Hash Algorithm Signature: RSA (1)
```
**tapdance2.freeaeskey.xyz**:
```
Cipher Suite: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)
Signature Hash Algorithm: 0x0601
    Signature Hash Algorithm Hash: SHA512 (6)
    Signature Hash Algorithm Signature: RSA (1)
```

**Comment by bvandersloot** [comment]
Surprisingly, no. I will resolve this.

**Comment by SergeyFrolov** [comment]
Fixed in #29 and #31
